_posts/2024-06-14-authz-openfga-introduction.md

```diff
@@ -0,0 +1,264 @@
+
+**Challenges with Authorization Model IDs**
+
+Another significant challenge is the management of Authorization Model IDs. 
+In OpenFGA, each model's ID is unknown until the model is created, and there's no way to set a predictable or semantic ID, such as a tag. 
+This lack of control over model IDs complicates the process of aligning and referencing the correct authorization model across multiple services.
+
+To mitigate this, we have developed a workaround: each service deployment includes a resource file specifying the used authorization model. 
+When a service starts, it compares the model detailed in its resource file against all models available on the OpenFGA server by iterating over them to find a match. 
+Although this method is currently sufficient, it is less than ideal, particularly at scale or in dynamic environments where models might frequently change.
+
+# SpringBoot Integration with OpenFGA
+
+In keeping with our tradition of providing practical examples, we have set up a SpringBoot integration with OpenFGA, which can be accessed in our GitHub repository. 
+This example serves as a straightforward guide for those looking to integrate OpenFGA into their SpringBoot applications.
+
+The complete example with the SpringBoot integration is available at: [AuthZ OpenFGA](https://github.com/sw-code/swcode-samples/tree/main/authz/authz-openfga)
+
+While we have extensively discussed the integration details in our previous articles with [Ory Keto]({% post_url 2023-10-13-authz-keto-introduction %}), 
+this example primarily focuses on swapping the Ory Keto API client with the OpenFGA client. 
+This change is straightforward due to the similarities in their relation-based authorization models which both systems derive from Google Zanzibar.
+
+The setup process remains consistent with our previous examples. 
+The project includes a docker-compose file that sets up the necessary infrastructure components, such as PostgreSQL, Keycloak, and OpenFGA.
+
+Upon starting the application, a specific Docker container is tasked with creating the authorization model in OpenFGA. 
+This setup ensures that everything needed to authenticate and authorize users is operational right from the get-go, allowing you to focus on the specific behaviors of your application rather than the underlying authentication and authorization mechanisms.
+
+## Why We Won't Dive Into Implementation Details
+Given the structural similarity to our previous example with [Ory Keto]({% post_url 2023-10-13-authz-keto-introduction %}), 
+we have chosen not to delve into the specifics of the codebase again. 
+The primary modification — replacing the Keto API client with the OpenFGA client — is minimal and straightforward, thus not necessitating a detailed walkthrough.
+
```

Comment:
it would be nice to add `the official [docs/guide](...) will be a great resource for anyone who wants to delve deeper`

(also linking is good fo SEO)